### PR TITLE
feat(alert): support callback on alert timeout

### LINKED ISSUE #4982: Callback for Alert timeout
**Describe the issue. What is the expected and unexpected behavior?**
When alert close after a certain timeout, there should be a callback to handle actions while closing(dismissing) alert, but there's no callback (or I'm unable to find one)

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**
Add value in array for alert, Alert shows properly and dismiss after timeout i.e 8sec but when alert dismisses we should have a callback if we want to do something on callback.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Enhancement

**What is your product and what release version are you targeting?**
Using patternfly/react-core: 4.32.1

### PR DESCRIPTION
**Pull request by boaz0**
**What**: Closes #4982 

//cc @QasimRRizvi

### CODE REVIEW

**Review by QasimRRizvi — APPROVED** [review]
LGTM

**Comment by boaz0** [comment]
Thanks @QasimRRizvi  for your review. :+1:

**Comment by boaz0** [comment]
Rebased the PR and now tests are green.

File: packages/react-integration/demo-app-ts/src/components/demos/AlertDemo/AlertCustomTimeoutDemo.tsx [hunk 1]
```diff
@@ -26,7 +27,7 @@ export class AlertCustomTimeoutDemo extends React.Component<{}, AlertCustomTimeo
           Open Alert
         </Button>
         {isOpenAlert && (
-          <Alert id="alert-custom-timeout" title="custom timeout" timeout={16000}>
+          <Alert onTimeout={this.onTimeout} id="alert-custom-timeout" title="custom timeout" timeout={16000}>
```
Comment (dlabrecq) [inline packages/react-integration/demo-app-ts/src/components/demos/AlertDemo/AlertCustomTimeoutDemo.tsx:30]: The alert should dismiss itself after the timeout, so setting isOpenAlert isn't necessary? The isOpenAlert appears to exist so the button can start the alert example.

**Review by dlabrecq — COMMENTED** [review]
Perhaps we could add to the existing "Alert Timeout" example? We show a JS `alert` for several alert examples already.

File: packages/react-integration/demo-app-ts/src/components/demos/AlertDemo/AlertCustomTimeoutDemo.tsx (hunk 1 above)
Comment (boaz0) [reply packages/react-integration/demo-app-ts/src/components/demos/AlertDemo/AlertCustomTimeoutDemo.tsx:30]: Yes, we can do that. But I am not sure cypress can catch `alert` messages.

Comment (boaz0) [reply packages/react-integration/demo-app-ts/src/components/demos/AlertDemo/AlertCustomTimeoutDemo.tsx:30]: Done

**Review by dlabrecq — APPROVED** [review]

**Review by redallen — APPROVED** [review]
Useful for tests, thanks!

**Pull request merged**